lib/multipart/post/composite_read_io.rb

```diff
@@ -24,6 +24,7 @@ def initialize(*ios)
 
       # Read from IOs in order until `length` bytes have been received.
       def read(length = nil, outbuf = nil)
+        raise ArgumentError, "The outbuf parameter must not be frozen" if !outbuf.nil? && outbuf.frozen?
         got_result = false
         outbuf = outbuf ? outbuf.replace("") : ""
```

Comment:
Doesn't `outbuf.replace("")` already break if `outbuf` is frozen? Do we really need an additional check?

Reply:
So this is precisely because it breaks but people might not know why. Here we explain that the outbuf parameter must not be frozen. It removes a layer of knowledge, and makes the behaviour explicit, rather than incidental.

Reply:
The only problem I have with this is the minor complexity and performance overhead, and we also fail if the user does the wrong thing.

Regarding why it exists, probably to match the interface of `IO#read`.